envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -62,23 +71,39 @@ internal class EnvoySnapshotFactory(
         return newSnapshotForGroup(group, globalSnapshot)
     }
 
-    private fun getServiceNamesForGroup(group: Group, globalSnapshot: Snapshot): List<String> {
+    private fun getRouteSpecificationsForGroup(group: Group, globalSnapshot: Snapshot): Collection<RouteSpecification> {
```

Comment:
Somehow the name is generic, but it deals only with services.
How about calling it `getServiceRouteSpecifications`?